src/core/dfn-finder.js

```diff
@@ -174,11 +176,18 @@ function getDfns(dfnForArray, parent, originalName, type) {
   }
   // Definitions that have a name and [data-dfn-for] that exactly match the
   // IDL entity:
-  const dfns = dfnForArray.filter(dfn => dfn.dataset.dfnFor === parent);
-  // If this is a top-level entity, and we didn't find anything with
-  // an explicitly empty [for], try <dfn> that inherited a [for].
+  const dfns = [];
+  for (const dfn of dfnForArray) {
+    /** @type {HTMLElement} */
+    const closestDfnFor = dfn.closest(`[data-dfn-for]`);
+    if (!closestDfnFor) {
+      continue;
+    }
+    if (closestDfnFor.dataset.dfnFor === parent) dfns.push(dfn);
+  }
```

Comment:
I prefer the reverse of b879864043a616057919492054ad1206a1b38bf8:

```js
const dfns = dfnForArray.filter(dfn =>
  dfn.closest(`[data-dfn-for="${parent}"]`)
)
```

Reply:
As long as the result is the same, I don’t mind. Feel free to update.